docs/source/repositories.md

```diff
@@ -0,0 +1,48 @@
+# Repositories Overview
+
+The **Leapp** project uses a modular repository structure in `repos/system_upgrade` to support multiple upgrade paths
+and shared functionality across these paths.
+Each repository contains actors, models, workflows,
+and supporting code for a specific upgrade path or common functionality.
+
+## High-Level Structure
+```
+repos/
+├── system_upgrade/
+│   ├── common/
+│   ├── el7toel8/
+│   ├── el8toel9/
+│   └── el9toel10
+│   └── ...
+```
+## Main Repositories
+
+- **`repos/system_upgrade/common/`**: Contains shared code that applies to all in-place upgrades,
+regardless of the specific source and target versions.
+- **`repos/system_upgrade/elXtoelY/`**: Contains code specific to concrete upgrade path.
```

Comment:
the upgrade path is usually understood as X.Y -> X1.Y1, where elXtoelY coveres all upgrade paths from X to Y. I would consider it ok, but some people could be confused by different wording / meaining in different parts of the documentation (and official RH docs). So instead of speaking about *upgrade path*, I would rather say something like:
```suggestion
- **`repos/system_upgrade/common/`**: Contains shared code that applies to all in-place upgrades,
regardless of the specific source and target versions.
- **`repos/system_upgrade/elXtoelY/`**: Contains code specific for in-place upgrades between EL X and EL Y systems.
```

Note that we do not want to use esplicitely RHEL in upstream documentation, as we would like to cover in future also CentoStream, and other forks using this project handles additional distributions. So going this more generic way seems to me better.